selfdrive/controls/lib/events.py

```diff
@@ -73,31 +74,37 @@ def create_alerts(self, event_types: List[str], callback_args=None):
       callback_args = []
 
     ret = []
-    for e in self.events:
+    for e, _ in self.events:
       types = EVENTS[e].keys()
       for et in event_types:
         if et in types:
           alert = EVENTS[e][et]
           if not isinstance(alert, Alert):
             alert = alert(*callback_args)
 
-          if DT_CTRL * (self.events_prev[e] + 1) >= alert.creation_delay:
+          if DT_CTRL * (self.events_t[e] + 1) >= alert.creation_delay:
             alert.alert_type = f"{EVENT_NAME[e]}/{et}"
             alert.event_type = et
             ret.append(alert)
     return ret
 
   def add_from_msg(self, events):
     for e in events:
-      self.events.append(e.name.raw)
+      meta = e.meta if len(e.meta) else None
```

Comment:
```suggestion
      meta = e.meta if e._has('meta') else None
```
Not sure if using this hidden method is cleaner